[conex] Add deprecation messages

--- conex/BUILD
+++ conex/BUILD
@@ -1,3 +1,5 @@
+_DEPRECATION = "DRAKE_DEPRECATED: The @conex external is deprecated in Drake's WORKSPACE and will be removed on or after 2023-11-01."
+
 cc_library(
     name = "test_util",
     srcs = glob(["test/test_util.cc"]),
@@ -111,6 +111,7 @@
     srcs = ["approximate_eigenvalues.cc"],
     hdrs = ["approximate_eigenvalues.h"],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = ["@eigen"],
 )
 
@@ -133,6 +134,7 @@
     ],
     hdrs = ["exponential_map_pade.h"],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = ["@eigen"],
 )
 
@@ -144,6 +146,7 @@
     ],
     hdrs = ["jordan_matrix_algebra.h"],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "@eigen",
     ],
@@ -175,6 +178,7 @@
     srcs = ["exponential_map.cc"],
     hdrs = ["exponential_map.h"],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "jordan_matrix_algebra",
         "@eigen",
@@ -241,6 +245,7 @@
         "tree_utils.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = ["@eigen"],
 )
 
@@ -277,6 +282,7 @@
         "workspace.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "tree_utils",
         "@eigen",
@@ -304,6 +310,7 @@
         "workspace.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "divergence",
         ":supernodal_solver",
@@ -321,6 +328,7 @@
         "linear_workspace.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "core",
         "@eigen",
@@ -340,6 +348,7 @@
         "psd_constraint.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "approximate_eigenvalues",
         "core",
@@ -362,6 +371,7 @@
         "newton_step.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = ["@eigen"],
 )
 
@@ -378,6 +388,7 @@
         "soc_constraint.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "core",
         "@eigen",
@@ -397,6 +408,7 @@
         "workspace.h",
     ],
     visibility = ["//visibility:public"],
+    deprecation = _DEPRECATION,
     deps = [
         "approximate_eigenvalues",
         "core",
