--- src/sys/objects/tagm.c
+++ src/sys/objects/tagm.c
@@ -158,12 +158,15 @@ PetscErrorCode  PetscCommDuplicate(MPI_Comm comm_in,MPI_Comm *comm_out,PetscMPII
     ierr = PetscInfo1(NULL,"Out of tags for object, starting to recycle. Comm reference count %" PetscInt_FMT "\n",counter->refcount);CHKERRQ(ierr);
     ierr = MPI_Comm_get_attr(MPI_COMM_WORLD,MPI_TAG_UB,&maxval,&flg);CHKERRMPI(ierr);
     if (!flg) SETERRQ(PETSC_COMM_SELF,PETSC_ERR_LIB,"MPI error: MPI_Comm_get_attr() is not returning a MPI_TAG_UB");
-    counter->tag = *maxval - 128; /* hope that any still active tags were issued right at the beginning of the run */
+    atomic_store(&counter->tag, *maxval - 128); /* hope that any still active tags were issued right at the beginning of the run */
   }
 
-  if (first_tag) *first_tag = counter->tag--;
+  if (first_tag) {
+    *first_tag = counter->tag;
+    atomic_fetch_sub(&counter->tag, 1);
+  }
 
-  counter->refcount++; /* number of references to this comm */
+  atomic_fetch_add(&counter->refcount, 1); /* number of references to this comm */
   ierr = PetscSpinlockUnlock(&PetscCommSpinLock);CHKERRQ(ierr);
   PetscFunctionReturn(0);
 }
