Add missing 'static' visibility annotation on private functions

--- Utilities/KWSys/vtksys/SystemTools.cxx
+++ Utilities/KWSys/vtksys/SystemTools.cxx
@@ -298,7 +298,7 @@
 typedef KWSYS_NAMESPACE::SystemTools::mode_t mode_t;
 #  endif
 
-inline int Mkdir(const std::string& dir, const mode_t* mode)
+static inline int Mkdir(const std::string& dir, const mode_t* mode)
 {
   int ret =
     _wmkdir(KWSYS_NAMESPACE::Encoding::ToWindowsExtendedPath(dir).c_str());
@@ -306,12 +306,12 @@
     KWSYS_NAMESPACE::SystemTools::SetPermissions(dir, *mode);
   return ret;
 }
-inline int Rmdir(const std::string& dir)
+static inline int Rmdir(const std::string& dir)
 {
   return _wrmdir(
     KWSYS_NAMESPACE::Encoding::ToWindowsExtendedPath(dir).c_str());
 }
-inline const char* Getcwd(char* buf, unsigned int len)
+static inline const char* Getcwd(char* buf, unsigned int len)
 {
   std::vector<wchar_t> w_buf(len);
   if (_wgetcwd(&w_buf[0], len)) {
@@ -329,12 +329,12 @@
   }
   return 0;
 }
-inline int Chdir(const std::string& dir)
+static inline int Chdir(const std::string& dir)
 {
   return _wchdir(KWSYS_NAMESPACE::Encoding::ToWide(dir).c_str());
 }
-inline void Realpath(const std::string& path, std::string& resolved_path,
-                     std::string* errorMessage = nullptr)
+static inline void Realpath(const std::string& path, std::string& resolved_path,
+                            std::string* errorMessage = nullptr)
 {
   std::wstring tmp = KWSYS_NAMESPACE::Encoding::ToWide(path);
   wchar_t* ptemp;
@@ -370,25 +370,25 @@
 
 #  include <fcntl.h>
 #  include <unistd.h>
-inline int Mkdir(const std::string& dir, const mode_t* mode)
+static inline int Mkdir(const std::string& dir, const mode_t* mode)
 {
   return mkdir(dir.c_str(), mode ? *mode : 00777);
 }
-inline int Rmdir(const std::string& dir)
+static inline int Rmdir(const std::string& dir)
 {
   return rmdir(dir.c_str());
 }
-inline const char* Getcwd(char* buf, unsigned int len)
+static inline const char* Getcwd(char* buf, unsigned int len)
 {
   return getcwd(buf, len);
 }
 
-inline int Chdir(const std::string& dir)
+static inline int Chdir(const std::string& dir)
 {
   return chdir(dir.c_str());
 }
-inline void Realpath(const std::string& path, std::string& resolved_path,
-                     std::string* errorMessage = nullptr)
+static inline void Realpath(const std::string& path, std::string& resolved_path,
+                            std::string* errorMessage = nullptr)
 {
   char resolved_name[KWSYS_SYSTEMTOOLS_MAXPATH];
 
