Use the Drake-compatible spelling of platforms (for Bazel 6.0)

--- tools/BUILD
+++ tools/BUILD
@@ -31,7 +31,7 @@
 
 config_setting(
     name = "apple",
-    constraint_values = ["@bazel_tools//platforms:osx"],
+    constraint_values = ["@platforms//os:osx"],
     visibility = ["//visibility:public"],
 )
 
@@ -58,7 +58,7 @@
 
 config_setting(
     name = "linux",
-    constraint_values = ["@bazel_tools//platforms:linux"],
+    constraint_values = ["@platforms//os:linux"],
     visibility = ["//visibility:public"],
 )
 
