--- src/sys/classes/viewer/interface/dlregispetsc.c
+++ src/sys/classes/viewer/interface/dlregispetsc.c
@@ -71,7 +71,6 @@ PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscmat(void);
 PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscdm(void);
 PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscksp(void);
 PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscsnes(void);
-PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscts(void);
   #endif

   /*
@@ -101,7 +100,6 @@ PETSC_EXTERN PetscErrorCode PetscDLLibraryRegister_petscsys(void)
   PetscCall(PetscDLLibraryRegister_petscdm());
   PetscCall(PetscDLLibraryRegister_petscksp());
   PetscCall(PetscDLLibraryRegister_petscsnes());
-  PetscCall(PetscDLLibraryRegister_petscts());
   #endif
   PetscFunctionReturn(0);
 }
