--- src/sys/objects/inherit.c
+++ src/sys/objects/inherit.c
@@ -3,6 +3,7 @@
 */
 #include <petsc/private/petscimpl.h> /*I   "petscsys.h"    I*/
 #include <petscviewer.h>
+#include <stdatomic.h>

 #if defined(PETSC_USE_LOG)
 PETSC_INTERN PetscObject *PetscObjects;
@@ -24,8 +25,8 @@ PETSC_EXTERN PetscErrorCode PetscObjectQueryFunction_Petsc(PetscObject, const ch

 PetscObjectId PetscObjectNewId_Internal(void)
 {
-  static PetscObjectId idcnt = 1;
-  return idcnt++;
+  static atomic_int atomic_idcnt = 1;
+  return atomic_fetch_add(&atomic_idcnt, 1);
 }

 /*
