--- src/sys/memory/mal.c
+++ src/sys/memory/mal.c
@@ -399,9 +399,15 @@ PetscErrorCode PetscMallocA(int n,PetscBool clear,int lineno,const char *functio
   if (petscmalloccoalesce) {
     char *p;
     PetscCall((*PetscTrMalloc)(sumbytes,clear,lineno,function,filename,(void**)&p));
-    for (i=0; i<n; i++) {
-      *ptr[i] = bytes[i] ? p : NULL;
-      p = (char*)PetscAddrAlign(p + bytes[i]);
+    if (p == NULL) {
+      for (i=0; i<n; i++) {
+        *ptr[i] = NULL;
+      }
+    } else {
+      for (i=0; i<n; i++) {
+        *ptr[i] = bytes[i] ? p : NULL;
+        p = (char*)PetscAddrAlign(p + bytes[i]);
+      }
     }
   } else {
     for (i=0; i<n; i++) {
