--- include/petsc/mpiuni/mpi.h
+++ include/petsc/mpiuni/mpi.h
@@ -153,8 +153,7 @@ typedef unsigned _int64          MPIUNI_UINT64;
  MPIUNI_ARG is used in the macros below only to stop C/C++ compilers
  from generating warning messages about unused variables while compiling PETSc.
 */
-MPIUni_PETSC_EXTERN void *MPIUNI_TMP;
-#define MPIUNI_ARG(arg) (MPIUNI_TMP = (void *)(MPI_Aint)(arg))
+#define MPIUNI_ARG(arg) ((void)0)

 #define MPI_IDENT     0
 #define MPI_CONGRUENT 1
