--- src/sys/objects/pname.c
+++ src/sys/objects/pname.c
@@ -131,7 +131,7 @@ PetscErrorCode  PetscObjectName(PetscObject obj)
     /* If the union has extra bytes, their value is implementation-dependent, but they will normally be what we set last
      * in 'ucomm.ptr = NULL'.  This output is always implementation-defined (and varies from run to run) so the union
      * abuse acceptable. */
-    ierr = PetscSNPrintf(name,64,"%s_%p_%D",obj->class_name,ucomm.ptr,counter->namecount++);CHKERRQ(ierr);
+    ierr = PetscSNPrintf(name,64,"%s_%p_%D",obj->class_name,ucomm.ptr,atomic_fetch_add(&counter->atomic_namecount,1));CHKERRQ(ierr);
     ierr = PetscStrallocpy(name,&obj->name);CHKERRQ(ierr);
   }
   PetscFunctionReturn(0);
