Do not load unused packages to reduce build footprint
--- src/sys/dll/reg.c
+++ src/sys/dll/reg.c
@@ -112,9 +112,6 @@ PETSC_INTERN PetscErrorCode PetscInitialize_DynamicLibraries(void)
 #if defined(PETSC_USE_SINGLE_LIBRARY)
   PetscCall(AOInitializePackage());
   PetscCall(PetscSFInitializePackage());
-#if !defined(PETSC_USE_COMPLEX)
-  PetscCall(CharacteristicInitializePackage());
-#endif
   PetscCall(ISInitializePackage());
   PetscCall(VecInitializePackage());
   PetscCall(MatInitializePackage());
@@ -123,7 +120,5 @@ PETSC_INTERN PetscErrorCode PetscInitialize_DynamicLibraries(void)
   PetscCall(KSPInitializePackage());
   PetscCall(SNESInitializePackage());
-  PetscCall(TSInitializePackage());
-  PetscCall(TaoInitializePackage());
 #else
   SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_SUP,"Cannot use -library_preload with multiple static PETSc libraries");
 #endif
