Do not load unused packages to reduce build footprint
--- src/sys/dll/reg.c
+++ src/sys/dll/reg.c
@@ -46,8 +46,6 @@ PETSC_EXTERN PetscErrorCode DMInitializePackage(void);
 PETSC_EXTERN PetscErrorCode PCInitializePackage(void);
 PETSC_EXTERN PetscErrorCode KSPInitializePackage(void);
 PETSC_EXTERN PetscErrorCode SNESInitializePackage(void);
-PETSC_EXTERN PetscErrorCode TSInitializePackage(void);
-PETSC_EXTERN PetscErrorCode TaoInitializePackage(void);
 #endif
 #if defined(PETSC_HAVE_THREADSAFETY)
 static MPI_Comm PETSC_COMM_WORLD_INNER = 0, PETSC_COMM_SELF_INNER = 0;
@@ -101,18 +99,11 @@ PETSC_INTERN PetscErrorCode PetscInitialize_DynamicLibraries(void)
     PetscCheck(found, PETSC_COMM_SELF, PETSC_ERR_FILE_OPEN, "Unable to locate PETSc KSP dynamic library \n You cannot move the dynamic libraries!");
     PetscCall(PetscLoadDynamicLibrary("snes", &found));
     PetscCheck(found, PETSC_COMM_SELF, PETSC_ERR_FILE_OPEN, "Unable to locate PETSc SNES dynamic library \n You cannot move the dynamic libraries!");
-    PetscCall(PetscLoadDynamicLibrary("ts", &found));
-    PetscCheck(found, PETSC_COMM_SELF, PETSC_ERR_FILE_OPEN, "Unable to locate PETSc TS dynamic library \n You cannot move the dynamic libraries!");
-    PetscCall(PetscLoadDynamicLibrary("tao", &found));
-    PetscCheck(found, PETSC_COMM_SELF, PETSC_ERR_FILE_OPEN, "Unable to locate Tao dynamic library \n You cannot move the dynamic libraries!");
   #endif
 #else /* defined(PETSC_HAVE_DYNAMIC_LIBRARIES) && defined(PETSC_USE_SHARED_LIBRARIES) */
   #if defined(PETSC_USE_SINGLE_LIBRARY)
     PetscCall(AOInitializePackage());
     PetscCall(PetscSFInitializePackage());
-    #if !defined(PETSC_USE_COMPLEX)
-    PetscCall(CharacteristicInitializePackage());
-    #endif
     PetscCall(ISInitializePackage());
     PetscCall(VecInitializePackage());
     PetscCall(MatInitializePackage());
@@ -120,8 +111,6 @@ PETSC_INTERN PetscErrorCode PetscInitialize_DynamicLibraries(void)
     PetscCall(PCInitializePackage());
     PetscCall(KSPInitializePackage());
     PetscCall(SNESInitializePackage());
-    PetscCall(TSInitializePackage());
-    PetscCall(TaoInitializePackage());
   #else
     SETERRQ(PETSC_COMM_WORLD, PETSC_ERR_SUP, "Cannot use -library_preload with multiple static PETSc libraries");
   #endif            

--- src/ksp/pc/impls/mpi/pcmpi.c
+++ src/ksp/pc/impls/mpi/pcmpi.c
@@ -395,8 +395,6 @@ PetscErrorCode PCMPIServerBegin(void)
     PetscCall(PCInitializePackage());
     PetscCall(KSPInitializePackage());
     PetscCall(SNESInitializePackage());
-    PetscCall(TSInitializePackage());
-    PetscCall(TaoInitializePackage());
   }

   PetscCallMPI(MPI_Comm_rank(PC_MPI_COMM_WORLD, &rank));

--- src/dm/interface/dmregall.c
+++ src/dm/interface/dmregall.c
@@ -55,7 +55,6 @@ PetscErrorCode DMRegisterAll(void)
 #if defined(PETSC_HAVE_MOAB)
   PetscCall(DMRegister(DMMOAB, DMCreate_Moab));
 #endif
-  PetscCall(DMRegister(DMNETWORK, DMCreate_Network));
   PetscCall(DMRegister(DMFOREST, DMCreate_Forest));
 #if defined(PETSC_HAVE_P4EST)
   PetscCall(DMRegister(DMP4EST, DMCreate_p4est));
