Remove redundant declarations to satisfy compilers.

--- absl/strings/internal/string_constant.h
+++ absl/strings/internal/string_constant.h
@@ -50,9 +50,6 @@ struct StringConstant {
                 "The input string_view must point to constant data.");
 };
 
-template <typename T>
-constexpr absl::string_view StringConstant<T>::value;  // NOLINT
-
 // Factory function for `StringConstant` instances.
 // It supports callables that have a constexpr default constructor and a
 // constexpr operator().
